src/main/java/org/quantumbadger/redreader/http/okhttp/OKHTTPBackend.java

```diff
@@ -31,11 +35,26 @@
 public class OKHTTPBackend implements HTTPBackend {
 
 	private final OkHttpClient mClient;
+	private static HTTPBackend httpBackend;
+	public static boolean reset = false;
```

Comment:
It might be better to move `reset` into `CacheDownload`, and add a static method called `CacheDownload.resetUserCredentialsOnNextRequest()` (so that the variable isn't public).

Also, requests are made from different threads, so this should probably be `volatile`.